lib/src/TailwindComponents/ButtonPureCss/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { Button as AriakitButton } from '@ariakit/react'
+
+import { classNames } from '../../classNames'
+
+import buttonStyles from './button.module.css'
+import type { ButtonProps } from './types'
+
+const cx = classNames(buttonStyles)
+
+export const ButtonPureCss = <T extends React.ElementType = 'button'>({
+  accessibleWhenDisabled = true,
+  children,
+  className = '',
+  disabled = false,
+  ref,
+  shape = 'default',
+  size = 'md',
+  variant = 'primary',
+  ...rest
+}: ButtonProps<T>) => {
+  const overrides = className.split(' ')
+
+  return (
+    <AriakitButton
+      {...rest}
+      accessibleWhenDisabled={accessibleWhenDisabled}
+      className={cx('root', `variant-${variant}`, `shape-${shape}`, `size-${size}`, ...overrides)}
```

Comment:
Is there a specific reason to split the classname and then spreading it?
Also there is no need to default the `className` property to an empty string as the utility already handle falsy parameters

Reply:
Next question is whether we need a utility at all or can we just do string concatenation ?

Reply:
Hum to answer your question @Leiksa, from what I read, splitting could prevent having duplicated classes?
It could also prevent us from having empty strings/ whitespaces:
const overrides = className.trim().split(/\s+/).filter(Boolean)

Not sure if it's relevant / useful.

What do you think?

Reply:
Ok for me, I played a little bit with it to understand better and now I got it 🙏